node/core/proposer/src/lib.rs

```diff
@@ -40,8 +40,11 @@ use sp_transaction_pool::TransactionPool;
 use prometheus_endpoint::Registry as PrometheusRegistry;
 use std::{fmt, pin::Pin, sync::Arc, time};
 
-/// How long proposal can take before we give up and err out
-const PROPOSE_TIMEOUT: core::time::Duration = core::time::Duration::from_millis(2500);
+/// How long propsing can take, before we give up and err out. We need a relatively large timeout
+/// here as long as we have large payload in statement distribution. Assuming we can reach most
+/// nodes within two hops, we will take about 4 seconds for transferring statemens of a size of
+/// about 5 Meg given validator bandwidth of 500 Mbit/s.
```

Comment:
Statements of size of about 5 Meg would take 200 ms with 400 Mbit/s and 2 hops, no? Do you mean 50 Meg? Or account for multiple peers and other traffic?

Reply:
This comment is kind of outdated, we might be able to reduce the time a bit again. Still the 200ms won't hold if multiple peers fetch from a single peer - which is going to happen. We limit this to ten at a time, so then we would be at 2 seconds. So I guess with something like 3 seconds we could be fine.